src/services/EcoBridge/OmniBridge/OmniBridge.reducers.ts

```diff
@@ -0,0 +1,155 @@
+import { TransactionReceipt } from '@ethersproject/abstract-provider'
+
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { TokenList } from '@uniswap/token-lists'
+
+import { BridgeDetails, BridgingDetailsErrorMessage, OmniBridgeList, SyncState } from '../EcoBridge.types'
+import { omniTransactionsAdapter } from './OmniBridge.adapter'
+import { InitialState, OmnibridgeTransactionMessage, OmniBridgeTxn } from './OmniBridge.types'
+
+const initialState: InitialState = {
+  transactions: omniTransactionsAdapter.getInitialState({}),
+  lists: {},
+  listsStatus: SyncState.IDLE,
+  bridgingDetails: {},
+  bridgingDetailsStatus: SyncState.IDLE,
+  lastMetadataCt: 0,
+}
+
+export const createOmniBridgeSlice = (bridgeId: OmniBridgeList) =>
+  createSlice({
+    name: bridgeId,
+    initialState,
+    reducers: {
+      addTransactions: (state, action: PayloadAction<OmniBridgeTxn[]>) => {
+        omniTransactionsAdapter.upsertMany(state.transactions, action.payload)
+      },
+      addTransaction: (state, action: PayloadAction<OmniBridgeTxn>) => {
+        const { payload: txn } = action
+
+        if (!txn.txHash) return
+
+        omniTransactionsAdapter.upsertOne(state.transactions, txn)
+      },
+      updateTransaction: (state, action: PayloadAction<{ txHash: string; receipt: TransactionReceipt }>) => {
+        const { receipt, txHash } = action.payload
+
+        omniTransactionsAdapter.updateOne(state.transactions, {
+          id: txHash,
+          changes: {
+            receipt,
+            timestampResolved: Date.now(),
+          },
+        })
+      },
+      setBridgeDetails: (state, action: PayloadAction<BridgeDetails>) => {
+        const { gas, fee, estimateTime, receiveAmount, requestId } = action.payload
+
+        //(store persist) crashing page without that code
+        if (!state.bridgingDetails) {
+          state.bridgingDetails = {}
+        }
+
+        if (requestId !== state.lastMetadataCt) {
+          if (state.bridgingDetailsStatus === SyncState.FAILED) return
+          state.bridgingDetailsStatus = SyncState.LOADING
+          return
+        } else {
+          state.bridgingDetailsStatus = SyncState.READY
+        }
+
+        state.bridgingDetails.gas = gas
+
+        state.bridgingDetails.fee = fee
+
+        state.bridgingDetails.estimateTime = estimateTime
+
+        state.bridgingDetails.receiveAmount = receiveAmount
```

Comment:
Not sure, but

```suggestion
        state.bridgingDetails.gas = { 
            gas,
            fee,
            estimateTime,
            receiveAmount
      }
```

Reply:
done [829f8c3](https://github.com/levelkdev/swapr-dapp/pull/1007/commits/829f8c30f9a20ae6b6c797db935bd09f2cad04db)